### PR TITLE
Fix two crash-and-burns

### PR DESCRIPTION
**Pull request by garbear**
Here's two fixes that i've been carrying for a while in my retroplayer branch

Isengard doesn't require the [dllLoader] fix, because this feature (using a function's ordinal value) isn't used.

EDIT: Added a third fix, `[addonlib]` (maybe not a crash-and-burn, but an unsafe cast is performed later)

### CODE REVIEW

File: xbmc/filesystem/ZipManager.cpp [hunk 1]
```diff
@@ -100,6 +100,16 @@ bool CZipManager::GetZipList(const CURL& url, std::vector<SZipEntry>& items)
   // Don't need to look in the last 18 bytes (ECDREC_SIZE-4)
   // But as we need to do overlapping between blocks (3 bytes),
   // we start the search at ECDREC_SIZE-1 from the end of file
+  if (fileSize == 0)
+  {
+    CLog::Log(LOGERROR, "ZipManager: Couldn't get zip file length");
+    return false;
```
Comment (arnova) [inline xbmc/filesystem/ZipManager.cpp]: What about (invalid) zip files which have 0 filesize? In that case the error message is not correct.

Comment (garbear) [inline xbmc/filesystem/ZipManager.cpp]: makes sense

**Comment by garbear** [comment]
Updated with arnova's suggestion

**Comment by Razzeee** [comment]
1 and 3 should be fine, can't really say anything about the dllLoader stuff.

**Comment by mkortstiege** [comment]
@Montellese, @Paxxi mind having a look? Should be ready to merge once the win32 commit is signed-off.

File: xbmc/cores/DllLoader/Win32DllLoader.cpp [hunk 2]
```diff
@@ -424,10 +426,14 @@ bool Win32DllLoader::ResolveOrdinal(const char *dllName, unsigned long ordinal,
 
 extern "C" FARPROC __stdcall dllWin32GetProcAddress(HMODULE hModule, LPCSTR function)
 {
-  // first check whether this function is one of the ones we need to wrap
-  void *fixup = NULL;
-  if (FunctionNeedsWrapping(win32_exports, function, &fixup))
-    return (FARPROC)fixup;
+  // if the high-order word is zero, then lpProcName is the function's ordinal value
+  if (reinterpret_cast<uintptr_t>(function) > std::numeric_limits<WORD>::max())
```
Comment (Paxxi) [inline xbmc/cores/DllLoader/Win32DllLoader.cpp]: I would probably go for >> 16 == 0 but this method works fine as well

Comment (garbear) [inline xbmc/cores/DllLoader/Win32DllLoader.cpp]: I preferred this way to avoid the magic number 16

Comment (Paxxi) [inline xbmc/cores/DllLoader/Win32DllLoader.cpp]: Both work and neither is 64-bit safe so doesn't matter :)

**Comment by Paxxi** [comment]
Ok with me

File: xbmc/cores/DllLoader/Win32DllLoader.cpp (hunk 2 above)
Comment (Paxxi) [inline xbmc/cores/DllLoader/Win32DllLoader.cpp]: Meh I'm getting rusty in my win32 api skills, this operation is common enough that there's a macro for it :) https://msdn.microsoft.com/en-us/library/windows/desktop/ms632657(v=vs.85).aspx

**Comment by mkortstiege** [comment]
jenkins build and merge.

File: xbmc/cores/DllLoader/Win32DllLoader.cpp (hunk 2 above)
Comment (garbear) [inline xbmc/cores/DllLoader/Win32DllLoader.cpp]: would have been a better choice. oh well :)

Comment (MartijnKaijser) [inline xbmc/cores/DllLoader/Win32DllLoader.cpp]: Could still PR it in ;)

**Pull request merged**